backend/src/services/journal/journal.service.js

```diff
@@ -8,17 +8,99 @@
  */
 
 class JournalService {
-  constructor(journalRepo) {
+  constructor(journalRepo, journalPermissions) {
     this.journalRepo = journalRepo;
+    this.journalPermissions = journalPermissions;
+  }
+
+  buildFiltersFromQuery(query, loggedInUser) {
+    const filters = {};
+
+    // Journal Ownership Filtering
+    let useDefaultUserFilter = true;
+    if (query.entire_class === true || query.entire_class === 'true') {
+      useDefaultUserFilter = false;
+    } else if (query.team_id != null) {
+      const teamId = parseInt(query.team_id, 10);
+      if (!isNaN(teamId)) {
+        filters.team_id = teamId;
+        useDefaultUserFilter = false;
+      }
+    } else if (query.user_id != null) {
+      const userId = parseInt(query.user_id, 10);
+      if (!isNaN(userId)) {
+        filters.user_id = userId;
+        useDefaultUserFilter = false;
+      }
+    }
+    // Filter by logged in user's id by default if no ownership filters provided
+    if (useDefaultUserFilter) {
+      filters.user_id = loggedInUser.id;
+    }
+
+    if (query.start_date) {
+      const startDate = new Date(query.start_date);
+      if (!isNaN(startDate.getTime())) {
+        filters.start_date = startDate;
+      }
+    }
+
+    if (query.end_date) {
+      const endDate = new Date(query.end_date);
+      if (!isNaN(endDate.getTime())) {
+        filters.end_date = endDate;
+      }
+    }
+
+    return filters;
+  }
+
+  mapFiltersToWhereClause(filters = {}) {
+    const where = {};
+
+    if (filters.user_id != null) {
+      where.user_id = filters.user_id;
+    }
+
+    if (filters.start_date != null) {
+      where.created_at = {};
+      where.created_at = { gte: filters.start_date };
+    }
+
+    if (filters.end_date != null) {
+      where.created_at = {};
+      where.created_at = { lte: filters.end_date };
+    }
+
```

Comment:
```suggestion
if (filters.start_date != null || filters.end_date != null) {
  where.created_at = {};
  if (filters.start_date != null) {
    where.created_at.gte = filters.start_date;
  }
  if (filters.end_date != null) {
    where.created_at.lte = filters.end_date;
  }
}
```
If both start_date and end_date are provided, the second one will override the first.